tools/compat/errno.rs

```diff
@@ -10,5 +10,9 @@ cfg_if::cfg_if! {
         pub unsafe fn errno_location() -> *mut c_int {
             libc::__error()
         }
+    } else if #[cfg(target_os = "android")] {
```

Comment:
Can you just fold this into the `macos` definition above?

Reply:
`macos` calls `__error`
`anroid` calls `__errno`

Reply:
Oh, oops, missed that.